src/KKTSolver/krylov.jl

```diff
@@ -110,5 +109,109 @@ function solve!(
     # Recover dx
     dx .= D * (kkt.A' * dy - ξd)
 
+    return nothing
+end
+
+
+# ==============================================================================
+#   KrylovSQDSolver: 
+# ==============================================================================
+
+"""
+    KrylovSQDSolver{T, F, Tv, Ta}
+
+Solves the augmented system with an iterative method `f::F`.
+
+```julia
+model.params.KKTOptions = Tulip.KKT.SolverOptions(
+    KrylovSQDSolver, method=Krylov.trimr
+)
+```
+"""
+mutable struct KrylovSQDSolver{T, F, Tv, Ta} <: KrylovSolver{T}
+    m::Int
+    n::Int
+
+    f::F
+
+    # Problem data
+    A::Ta     # Constraint matrix
+    θ::Tv     # scaling
+    regP::Tv  # primal regularization
+    regD::Tv  # dual regularization
+
+    function KrylovSQDSolver(f::Function, A::Ta) where{Ta <: AbstractMatrix}
+        F = typeof(f)
+        m, n = size(A)
+        T = eltype(A)
+        return new{T, F, Vector{T}, Ta}(
+            m, n,
+            f,
+            A, ones(T, n), ones(T, n), ones(T, m)
+        )
+    end
+end
+
+setup(::Type{KrylovSQDSolver}, A; method=Krylov.trimr) = KrylovSQDSolver(method, A)
+
+backend(kkt::KrylovSQDSolver) = "Krylov ($(kkt.f))"
+linear_system(::KrylovSQDSolver) = "Augmented system"
+
+
+"""
+    update!(kkt::KrylovSQDSolver, θ, regP, regD)
+
+Update diagonal scaling ``θ`` and primal-dual regularizations.
+"""
+function update!(
+    kkt::KrylovSQDSolver{Tv},
+    θ::AbstractVector{Tv},
+    regP::AbstractVector{Tv},
+    regD::AbstractVector{Tv}
+) where{Tv<:Real}
+    # Sanity checks
+    length(θ)  == kkt.n || throw(DimensionMismatch(
+        "θ has length $(length(θ)) but linear solver is for n=$(kkt.n)."
+    ))
+    length(regP) == kkt.n || throw(DimensionMismatch(
+        "regP has length $(length(regP)) but linear solver has n=$(kkt.n)"
+    ))
+    length(regD) == kkt.m || throw(DimensionMismatch(
+        "regD has length $(length(regD)) but linear solver has m=$(kkt.m)"
+    ))
+
+    # Update diagonal scaling
+    kkt.θ .= θ
+    # Update regularizers
+    kkt.regP .= regP
+    kkt.regD .= regD
+
+    return nothing
+end
+
+"""
+    solve!(dx, dy, kkt::KrylovSQDSolver, ξp, ξd)
+
+Solve the normal equations system using an iterative method.
+"""
+function solve!(
+    dx::Vector{Tv}, dy::Vector{Tv},
+    kkt::KrylovSQDSolver{Tv},
+    ξp::Vector{Tv}, ξd::Vector{Tv}
+) where{Tv<:Real}
+    m, n = kkt.m, kkt.n
+
+    # Setup
+    d = one(Float64) ./ (kkt.θ .+ kkt.regP)
+    D = LO.opDiagonal(d)
```

Comment:
The culprit is here, `opDiagonal(d)` allocates a new vector at each operator-vector product.